doc/asynchronous_fifo/asynchronous_fifo.rst

```diff
@@ -84,7 +84,7 @@ The Asynchronous FIFO has the following functions to control the FIFO:
 
 * ``asynchronous_fifo_consumer_get()`` gets one sample from the FIFO. It
   must be given a timestamp related to when this (or the previous) sample
-  is (was) output.
+  is (was) output. It returns 1 if the pulled samples are valid else zero.
```

Comment:
You could future proof (and meet standard convention) by changing something like "returns non-zero for error, otherwise zero". Opens if to adding other return error values going forward.

Reply:
I went round this loop a few times but yes, you're right it would be helpful, perhaps with an enum. Say, knowing reset vs underflow might be good. I'll update.